Common/Data/Market/DataDictionary.cs

```diff
@@ -12,10 +12,17 @@ public class DataDictionary<T> : IDictionary<string, T>
         // storage for the data
         private readonly IDictionary<string, T> _data = new Dictionary<string, T>();
 
+        /// <summary>
+        /// Empty Constructor
+        /// </summary>
         public DataDictionary()
         {
         }
 
+        /// <summary>
+        /// Constructor which sets teh Time to the frontier
+        /// </summary>
+        /// <param name="frontier"></param>
```

Comment:
Can we add comments for the parameters?  Adding them like this removes the compiler warning making it harder to find again.